code/modules/heavy_vehicle/equipment/utility.dm

```diff
@@ -4,46 +4,51 @@
 	icon_state = "mecha_clamp"
 	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
 	restricted_software = list(MECH_SOFTWARE_UTILITY)
-	var/obj/carrying
+	var/carrying_capacity = 5
+	var/list/obj/carrying = list()
 	origin_tech = list(TECH_MATERIAL = 2, TECH_ENGINEERING = 2)
 
 /obj/item/mecha_equipment/clamp/resolve_attackby(atom/A, mob/user, click_params)
 	if(istype(A, /obj/structure/closet) && owner)
-		return 0
+		return FALSE
 	return ..()
 
 /obj/item/mecha_equipment/clamp/attack_hand(mob/user)
 	if(owner && LAZYISIN(owner.pilots, user))
-		if(!owner.hatch_closed && carrying)
-			if(user.put_in_active_hand(carrying))
-				owner.visible_message("<span class='notice'>\The [user] carefully grabs \the [carrying] from \the [src].</span>")
-				carrying = null
+		if(!owner.hatch_closed && length(carrying))
+			var/obj/chosen_obj = input(user, "Choose an object to grab.", "Clamp Claw") as null|anything in carrying
+			if(!chosen_obj)
+				return
+			if(user.put_in_active_hand(chosen_obj))
+				owner.visible_message(SPAN_NOTICE("\The [user] carefully grabs \the [chosen_obj] from \the [src]."))
+				carrying -= chosen_obj
```

Comment:
What I'd actually do here is:
- make the input happen when you alt click;
- when you choose an object and you lay it down, it automatically puts the other object as the chosen object.

This way it's much easier to put down a large amount of crates.

Reply:
donezo